tests/test-integration/src/test/java/org/eclipse/jetty/test/FailedSelectorTest.java

```diff
@@ -0,0 +1,390 @@
+        handlers.addHandler(context);
+        handlers.addHandler(new DefaultHandler());
+
+        server.setHandler(handlers);
+
+        server.start();
+    }
+
+    @Test
+    public void testRestartServerOnSelectFailure() throws Exception
+    {
+        CountDownLatch failedLatch = new CountDownLatch(1);
+
+        startServer((server) ->
+        {
+            RestartSelectorCustomConnector connector = new RestartSelectorCustomConnector(server, 1, 1, new RestartServerTask(server, failedLatch));
+            connector.setPort(0);
+            connector.setIdleTimeout(1000);
+            return connector;
+        });
+
+        // Request /hello
+        assertRequestHello();
+
+        // Request /selector/close
+        assertRequestSelectorClose("/selector/close");
+
+        // Wait for selectors to close from action above
+        assertTrue(failedLatch.await(2, TimeUnit.SECONDS));
+
+        // Request /hello
+        assertRequestHello();
+    }
+
+    @Test
+    public void testRestartSelectorOnSelectFailure() throws Exception
+    {
+        CountDownLatch failedLatch = new CountDownLatch(1);
+
+        startServer((server) ->
+        {
+            RestartSelectorCustomConnector connector = new RestartSelectorCustomConnector(server, 1, 1, new RestartSelectorTask(failedLatch));
+            connector.setPort(0);
+            connector.setIdleTimeout(1000);
+            return connector;
+        });
+
+        // Request /hello
+        assertRequestHello();
+
+        // Request /selector/close
+        assertRequestSelectorClose("/selector/close");
+
+        // Wait for selectors to close from action above
+        assertTrue(failedLatch.await(2, TimeUnit.SECONDS));
+
+        // Request /hello
+        assertRequestHello();
+    }
+
+    private void assertRequestSelectorClose(String path) throws InterruptedException, ExecutionException, TimeoutException
+    {
+        URI dest = server.getURI().resolve(path);
+        LOG.info("Requesting GET on {}", dest);
+
+        ContentResponse response = client.newRequest(dest)
+            .method(HttpMethod.GET)
+            .header(HttpHeader.CONNECTION, "close")
+            .send();
+
+        assertThat("/selector/close status", response.getStatus(), is(HttpStatus.OK_200));
+        assertThat("/selector/close response", response.getContentAsString(), startsWith("Closing selectors "));
+    }
+
+    private void assertRequestHello() throws InterruptedException, ExecutionException, TimeoutException
+    {
+        URI dest = server.getURI().resolve("/hello");
+        LOG.info("Requesting GET on {}", dest);
+        ContentResponse response = client.newRequest(dest)
+            .method(HttpMethod.GET)
+            .header(HttpHeader.CONNECTION, "close")
+            .send();
+
+        assertThat("/hello status", response.getStatus(), is(HttpStatus.OK_200));
+        assertThat("/hello response", response.getContentAsString(), startsWith("Hello "));
```

Comment:
Refactor the `/hello` string to be a variable, and use it in both `URI.resolve()` at the beginning of the method, and in the asserts.

Same should be done for `assertRequestSelectorClose()`.

Reply:
Done.